### DISCUSSION TITLE
how to build my own prefetcher

### DISCUSSION THREAD

**Discussion by M11a23x58** [ChampSim/ChampSim#340] [Q&A]
there are 5 functions in no.cc
prefetcher_initialize
prefetcher_cache_operate
prefetcher_cache_fill
prefetcher_cycle_operate
prefetcher_final_stats
I dont know what these functions should do.
Anyone can help me?

**Comment by ngober** [comment]
Each of them are called when different events happen:

- `prefetcher_initialize` is called at the beginning of the simulation
- `prefetcher_finial_stats` is called at the end of the simulation
- `prefetcher_cache_operate` is called whenever a demand triggers a tag check (whether it is a hit or a miss)
- `prefetcher_cache_fill` is called whenever a miss is filled.
- `prefetcher_cycle_operate` is called once per cycle
